compiler/src/dotty/tools/dotc/classpath/DirectoryClassPath.scala

```diff
@@ -286,7 +286,7 @@ case class DirectoryClassPath(dir: JFile) extends JFileDirectoryLookup[ClassFile
   }
 
   protected def createFileEntry(file: AbstractFile): ClassFileEntryImpl = ClassFileEntryImpl(file)
-  protected def isMatchingFile(f: JFile): Boolean = f.isClass
+  protected def isMatchingFile(f: JFile): Boolean = f.isClass || f.isTasty
```

Comment:
I think we should be careful here, because at least with the Zip based class paths including tasty doubles the memory usage

Reply:
This is still a work in progress.

Currently, we list `.class` and then when we parse them we load the `.tasty` file, doubling the memory usage at that point. The aim is to get to a point where we only need to read the `.tasty` and do not have to load the `.class`.

Here is where we currently load the `.tasty`: https://github.com/lampepfl/dotty/blob/2e193043749ca243025cd2ebc3ae8d5faff17b31/compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala#L951-L974

Reply:
FWIW, tasty-query does it the other way around: it first looks for a .tasty file. If there is one, it loads it and disregards any associated .class file. It only loads .class files as backup for not finding tasty (for Java and Scala 2).